optimum/intel/ipex/modeling_base.py

```diff
@@ -319,6 +347,8 @@ def _init_warmup(self):
         if not self._is_ipex_exported:
             use_cache = "past_key_values" in self.input_names
             dummy_inputs = prepare_jit_inputs(self, self.export_feature, use_cache)
+            if "cpu" not in str(self._device):
+                dummy_inputs = {name: tensor.to(self._device) for name, tensor in dummy_inputs.items()}
```

Comment:
To also make it work for inputs which are list or tuple (like `past_key_values`) you need to add a [`recursive_to_device`]((https://github.com/huggingface/optimum/blob/b3ecb6c405b7fd5425d79483fd7dc88c0609be8e/optimum/exporters/onnx/utils.py#L120)) function (can be added [here](https://github.com/huggingface/optimum-intel/blob/main/optimum/intel/utils/)) and then : 

```python
for key, value in dummy_inputs.items():
    dummy_inputs[key] = recursive_to_device(value=value, device=self._device)
```

Reply:
Thx for the suggestion! I added a `recursive_to_device` function, which also works for a dictionary. So we can save `for key, value in dummy_inputs.items():`. Pls have a review.